### PR TITLE
Update tests, and add documentation!

### PR DESCRIPTION
**Pull request by strega-nil**
This PR does the following:

  - fix tests -- now, they're always built in the CMake scripts, and they work on VS2015
    - add a new flag, `BUILD_TESTING`, which allows one to turn off testing builds
  - Add documentation for running tests

### CODE REVIEW

File: docs/tool-maintainers/testing.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+```
+
+Now build the tests again, and run them:
+
+```sh
+cmake --build .
+./vcpkg-test
+```
+
+Hopefully, that worked! It should compile correctly, and have one failing test.
```
Comment (vicroms) [inline docs/tool-maintainers/testing.md]: Show an example of the output produced when running the tests.

File: docs/tool-maintainers/testing.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+```
+
+Now compile and build the tests, and this time let's only run our example tests:
+
+```sh
+cmake --build .
+./vcpkg-test [example]
+```
+
+Hopefully you have one test running and succeeding! If you have that, you have
```
Comment (vicroms) [inline docs/tool-maintainers/testing.md]: Also example output here of what a successful run looks like.

**Review by vicroms — COMMENTED** [review]
LGTM

**Comment by ubsan** [comment]
thanks y'all!

File: docs/tool-maintainers/testing.md [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+tag all of the test cases with `[arguments]`.
+
+If you wish to add helper functions, make sure to place them in an anonymous
+namespace -- this will ensure that they don't trample over anybody else's
+space. Additionally, there are a few helper functions that live in
+`<vcpkg-test/util.h>` and `src/vcpkg-test/util.cpp` -- make sure to look into
+them so that you're not rewriting functionality.
+
+That should be all you need to know to start writing your own tests!
+Remember to check out the [Catch2 documentation]
```
Comment (grdowns) [inline docs/tool-maintainers/testing.md]: TODO insert the link here

**Pull request merged**